docs: document the SDKs approach to encoding (#2405)
FuelLabs/fuels-ts: apps/docs/src/guide/encoding/index.md

# Encoding

Encoding is the process of serializing data into a format that is suitable for transmission or storage. This is important for blockchains as it enables state minimization and efficiency, as well as for generating proofs.

> Fortunately if you are working with [program types](https://docs.fuel.network/docs/sway/sway-program-types/) on the Fuel network such as [calling contracts](https://docs.fuel.network/docs/fuels-ts/contracts/), the SDK will handle encoding for you automatically and it will adhere to the [argument encoding specification](https://docs.fuel.network/docs/specs/abi/argument-encoding/), so you are free to work with data in it's expected format rather than as bytecode.

Reviewer: Nits.

```suggestion
Fortunately, if you are working with [program types](https://docs.fuel.network/docs/sway/sway-program-types/) on the Fuel network such as [calling contracts](https://docs.fuel.network/docs/fuels-ts/contracts/), the SDK will handle encoding for you automatically. It will adhere to the [argument encoding specification](https://docs.fuel.network/docs/specs/abi/argument-encoding/), so you can work with data in its expected format rather than as bytecode.
```

> However there may be scenarios where you want to manipulate call or return data yourself, or even implement your own serialization specification. This guide will cover how to encode and decode data using the SDK.

Reviewer: Nits.

```suggestion
However, there may be scenarios where you want to manipulate call or return data yourself or even implement your own serialization specification. This guide will cover how to encode and decode data using the SDK.
```
